tests/api/program/test_code_generation.py

```diff
@@ -75,15 +65,10 @@ def test_code_generation_with_empty_program():
     with pq.Program() as program:
         pass
 
-    code = program.as_code()
+    code = program._as_code()
```

Comment:
I would rewrite these tests to use the public `as_code` function, or simply delete these tests, since it tests a private method.